R/swagger.R

```diff
@@ -191,5 +346,58 @@ removeNaOrNulls <- function(x) {
   ret <- lapply(x, removeNaOrNulls)
   class(ret) <- class(x)
 
+  # Put example back in
+  if (saveExample) {
+    ret$example <- savedExample
+  }
+
   ret
 }
+
+#' For openapi definition
+#' @noRd
+priorizeProperty <- function(...) {
+  l <- list(...)
+  isnullordefault <- sapply(l, function(x) {isNaOrNull(x) || isTRUE(attributes(x)$default)})
+  l[[which.min(isnullordefault)]]
```

Comment:
I've been tripped up many times. `which.min(logical_vect)` does not return the first value. It returns the smallest valid location, which is `FALSE`, a falsey value.

```r
❯❯ sample(21:25) %>% print() %>% which.min()
[1] 25 23 22 24 21
[1] 5

❯❯ sample(c(T,T,F,F,F)) %>% print() %>% which.min()
[1]  TRUE FALSE FALSE  TRUE FALSE
[1] 2
# Not expected!

❯❯ sample(c(T,T,F,F,F)) %>% print() %>% which() %>% min()
[1] FALSE FALSE FALSE  TRUE  TRUE
[1] 4
# Expected!
```

```suggestion
  l[[min(which(isnullordefault))]]
```

Reply:
Are we guaranteed that there is a valid value?  I don't want `which()` to return a `numeric(0)`

Reply:
> ```r
> sample(c(T,T,F,F,F)) %>% print() %>% which() %>% min()
> ```
As I understand it FALSE is 0 and TRUE is 1, which.min will return first FALSE, then TRUE.
For this particular function, it needs to return like your sample 2

Reply:
To make it clear, we are looking for the first value in the list where the value from the expression above is FALSE.

`min(which())` is going to return the first position of a TRUE value, which is not what this function is looking for.

I'll attempt a simpler prioritizer.

Reply:
> 
> 
> Are we guaranteed that there is a valid value? I don't want `which()` to return a `numeric(0)`

It can if the function is called without parameters. Will fix.